compiler/AST/view.cpp

```diff
@@ -1154,6 +1226,30 @@ void set_view(std::set<BlockStmt*>& bss) {
   }
 }
 
+//
+// typesWithName: print all TypeSymbols with the given name
+//
+
+void typesWithName(const char* name) {
+  typesWithName(name, gTypeSymbols);
+}
+
+void typesWithName(const char* name, Vec<TypeSymbol*>& tyVec) {
+  printf("typesWithName(\"%s\")\n", name);
+  int count = 0, countNonNull = 0;
+  forv_Vec(TypeSymbol, ty, tyVec) {
+    if (ty) {
+      countNonNull++;
+      if (!strcmp(ty->name, name)) {
+        count++;
+        printf("  sym %8d   %s %d  %s\n", ty->id, ty->type->astTagAsString(),
+               ty->type->id, debugLoc(ty));
+      }
+    }
+  }
+  printf("  = %d type(s) of %d\n", count, countNonNull);
+}
```

Comment:
there's a lot of debug code in here that would be better in another PR

Reply:
I don't think it'd be "trivial" but I'd expect it would be pretty easy to get it reviewed